Feat/16290/add button component (#16305)
MetaMask/metamask-extension: ui/components/component-library/button/README.mdx

import { Story, Canvas, ArgsTable } from '@storybook/addon-docs';
import { Button } from './button';

# Button

The `Button` is used for user actions it unifies `ButtonPrimary`, `ButtonSecondary` and `ButtonLink`

<Canvas>
  <Story id="ui-components-component-library-button-button-stories-js--default-story" />
</Canvas>

## Props

> The `Button` accepts all props below as well as all [Box](/docs/ui-components-ui-box-box-stories-js--default-story#props) component props

Reviewer: Let's add that it accepts all prop for the other button types as well

<ArgsTable of={Button} />

### Type

Use the `type` prop and the `BUTTON_TYPES` object from `./button.constants.js` to change the `Button` type.

Possible types include:

- `BUTTON_TYPES.PRIMARY`
- `BUTTON_TYPES.SECONDARY`
- `BUTTON_TYPES.LINK`

<Canvas>
  <Story id="ui-components-component-library-button-button-stories-js--type" />
</Canvas>

```jsx
import { Button, BUTTON_TYPES } from '../ui/component-library/button';

> <Button type={BUTTON_TYPES.PRIMARY}>Primary Button</Button>
> <Button type={BUTTON_TYPES.SECONDARY}>Secondary Button</Button>
> <Button type={BUTTON_TYPES.LINK}>Link Button</Button>

Reviewer: To be consistent with out naming should we just use the names verbatim?
```suggestion
<Button type={BUTTON_TYPES.PRIMARY}>Button Primary</Button>
<Button type={BUTTON_TYPES.SECONDARY}>Button Secondary</Button>
<Button type={BUTTON_TYPES.LINK}>Button Link</Button>
```

import { Story, Canvas, ArgsTable } from '@storybook/addon-docs';
import { Button } from './button';

> ### This is a base component. It should not be used in your feature code directly but as a "base" for other UI components

Reviewer: Remove this
```suggestion
```

# Button

> The `Button` is the base component for buttons.

Reviewer: Update the description

```suggestion
The `Button` is used for user actions it unifies `ButtonPrimary`, `ButtonSecondary` and `ButtonLink`
```
